tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -640,6 +645,15 @@ def fit(self,
         ag_args_ensemble = kwargs['ag_args_ensemble']
         excluded_model_types = kwargs['excluded_model_types']
 
+        quantile_levels = kwargs.get('quantile_levels', None)
+        if isinstance(quantile_levels, float):
+            quantile_levels = [quantile_levels]
+        if isinstance(quantile_levels, Iterable):
+            quantile_levels = np.sort(np.array(quantile_levels))
+        self._learner.quantile_levels = quantile_levels
+        if quantile_levels is not None and self._learner.problem_type is None:
+            self._learner.problem_type = QUANTILE
+
```

Comment:
I'm not sure this is 100% safe. What if user erroneously specifies `quantile_levels` but actually has a classification task?

I think if `problem_type` is originally None, then it should be first inferred as autogluon did before. At that point if `problem_type != REGRESSION`, and `quantile_levels is not None`, we should raise ValueError("autogluon infers this to be classification problem for which `quantile_levels` cannot be specified. If it is truly a quantile regression problem, please specify: `problem_type='quantile'`"). 

Otherwise if the inferred problem_type==REGRESSION and quantile_levels is not None, then you can overwrite problem_type = QUANTILE.

Reply:
That way, should it be better to fix this at
[tabular/src/autogluon/tabular/learner/default_learner.py#L118](https://github.com/awslabs/autogluon/blob/master/tabular/src/autogluon/tabular/learner/default_learner.py#L118)
```suggestion
       # default_learner.py#L118
        if self.problem_type is None:
            self.problem_type = self.infer_problem_type(X[self.label])            
            if quantile_levels is not None:
                if self.problem_type == REGRESSION:
                    self.problem_type  = QUANTILE
                else:
                     raise ValueError("autogluon infers this to be classification problem for which quantile_levels cannot be 
                                                   specified. If it is truly a quantile regression problem, please 
                                                   specify:problem_type='quantile'")
```